src/array.c

```diff
@@ -150,7 +152,17 @@ SDB_API int sdb_array_insert(Sdb *s, const char *key, int idx, const char *val,
 	// we can optimize this by caching value len in memory . add
 	// sdb_const_get_size()
 	lstr = strlen (str); 
-	x = malloc (lval + lstr + 2);
+
+	// When removing strlen this conversion should be checked
+	size_t lstr_tmp = lstr;
+	if (lstr_tmp > INT_MAX - lval || lstr_tmp + lval > INT_MAX - 2) {
```

Comment:
in r2land we use those macros, maybe better to use macros instead of typing that everytime
```
/* preventive math overflow checks */
#define UT64_ADD_OVFCHK(x,y) ((UT64_MAX - x) <= y)
#define UT32_ADD_OVFCHK(x,y) ((UT32_MAX - x) <= y)
#define UT16_ADD_OVFCHK(x,y) ((UT16_MAX - x) <= y)
#define UT8_ADD_OVFCHK(x,y) ((UT8_MAX - x) <= y)
```